// nuxi/dist/chunks/build.mjs
import { r as resolve } from '../shared/nuxi-edge.a03f3b10.mjs';
import { c as consola } from '../shared/nuxi-edge.5f813239.mjs';
import { w as writeTypes } from '../shared/nuxi-edge.9202ce8d.mjs';
import { l as loadKit } from '../shared/nuxi-edge.b94eae5e.mjs';
import { c as clearDir } from '../shared/nuxi-edge.3196727a.mjs';
import { o as overrideEnv } from '../shared/nuxi-edge.2b130854.mjs';
import { d as defineNuxtCommand } from '../shared/nuxi-edge.f39aafd9.mjs';
import '../shared/nuxi-edge.76bb3f1e.mjs';
import 'util';
import 'path';
import 'fs';
import 'os';
import 'tty';
import 'node:fs';
import 'node:module';
import 'node:url';

const buildCommand = defineNuxtCommand({
  meta: {
    name: "build",
    usage: "npx nuxi build [--prerender] [rootDir]",
    description: "Build nuxt for production deployment"
  },
  async invoke(args) {
    overrideEnv("production");
    const rootDir = resolve(args._[0] || ".");
    const { loadNuxt, buildNuxt } = await loadKit(rootDir);
    const nuxt = await loadNuxt({
      rootDir,
      overrides: {
        _generate: args.prerender
      }
    });
    await clearDir(nuxt.options.buildDir);
    await writeTypes(nuxt);
    nuxt.hook("build:error", (err) => {
      consola.error("Nuxt Build Error:", err);
      process.exit(1);
    });
    await buildNuxt(nuxt);
  }
});

export { buildCommand as default };
